File: PrimeFactoringBenchmark.jl
using Primes
using Random

### Number Of Primes
x = 19

primeNumbers = Primes.primes((2^x))
compositeNumbers = primeNumbers.*primeNumbers[randperm(length(primeNumbers))]
factors = zeros(length(primeNumbers),2)


t1 = time()
for idx = 1:length(compositeNumbers)
    factors[idx,:] = factor(Vector, compositeNumbers[idx])
end
t2 = time()

println("Seconds Taken: ", (t2-t1))

File: PrimeFactoringBenchmark.m
x = 19;
for i = 1:6
    primeNumbers = primes(uint64(2^x));
    compositeNumbers = primeNumbers.*primeNumbers(randperm(numel(primeNumbers)));
    factors = zeros(numel(primeNumbers),2);

    tic;
    parfor idx = 1:numel(compositeNumbers)
        factors(idx,:) = factor(compositeNumbers(idx));
    end
    toc
end